test/functional/services/test_subjects.js

```diff
@@ -51,19 +54,57 @@ export function TestSubjectsProvider({ getService }) {
 
     async setValue(selector, text) {
       return await retry.try(async () => {
-        const input = await this.find(selector);
-        await input.click();
+        const element = await this.find(selector);
+        await element.click();
+
+        // in case the input element is actually a child of the testSubject, we
+        // call clearValue() and type() on the element that is focues after
+        // clicking on the testSubject
+        const input = await remote.getActiveElement();
         await input.clearValue();
         await input.type(text);
       });
     }
 
+    async isEnabled(selector) {
+      return await retry.try(async () => {
+        const element = await this.find(selector);
+        return await element.isEnabled();
+      });
+    }
+
+    async isSelected(selector) {
+      return await retry.try(async () => {
+        const element = await this.find(selector);
+        return await element.isSelected();
+      });
+    }
+
+    async isSelectedAll(selectorAll) {
+      return await this._mapAll(selectorAll, async (element) => {
+        return await element.isSelected();
+      });
+    }
+
     async getVisibleText(selector) {
       return await retry.try(async () => {
         const element = await this.find(selector);
         return await element.getVisibleText();
       });
     }
+
+    async getVisibleTextAll(selectorAll) {
+      return await this._mapAll(selectorAll, async (element) => {
+        return await element.getVisibleText();
+      });
+    }
+
+    async _mapAll(selectorAll, mapFn) {
```

Comment:
Huh, I would have sworn that this was already part of the Styleguide. I'm pretty firmly against the reasoning at the link, and thought we had already resolved this... If you want to open the vote issue I'm fine with it, but I've been using underscored methods on class since I started using the class syntax...